system_setup/server/controllers/configure.py

```diff
@@ -40,89 +41,189 @@ def __init__(self, app):
-                    raise SystemError('Command {} failed with return code {}'
-                                      .format(cp.args, cp.returncode))
+
+            return True
+
+        cache.update()
+        cache.open(None)
+        with cache.actiongroup():
+            for package in packages:
+                cache[package].mark_install()
+
+            cache.commit()
+
+        return True
+
+    def _update_locale_gen_file(self, localeGenPath, lang) -> bool:
+        """ Uncomments the line in locale.gen file where lang is found,
+            if found commented. A fully qualified language is expected,
+            since that would have passed thru the Locale controller
+            validation. e.g. en_UK.UTF-8. Returns True for success. """
+
+        fileContents: str
+        try:
+            with open(localeGenPath, "r+") as f:
+                fileContents = f.read()
+                lineFound = fileContents.find(lang)
+                if lineFound == -1:
+                    # An unsupported locale coming from our UI is a bug
+                    log.error("Selected language %s not supported.", lang)
+                    return False
+
+                pattern = r'[# ]*({}.*)'.format(lang)
+                commented = re.compile(pattern)
+                (fileContents, n) = commented.subn(r'\1', fileContents,
+                                                   count=1)
+                if n != 1:
+                    log.error("Unexpected locale.gen file contents. Aborting.")
+                    return False
+
+                f.seek(0)
+                f.write(fileContents)
+                return True
+
+        except IOError as err:
+            log.error("Failed to modify %s file. %s", localeGenPath, err)
+            return False
+
+    def _locale_gen_file_path(self):
+        localeGenPath = "/etc/locale.gen"
+        if self.app.opts.dry_run is False or self.app.opts.dry_run is None:
+            return localeGenPath
+
+        # For testing purposes.
+        etc_dir = os.path.join(self.model.root, "etc")
+        testLocGenPath = os.path.join(etc_dir,
+                                      os.path.basename(localeGenPath))
+        shutil.copy(localeGenPath, testLocGenPath)
+        shutil.copy(localeGenPath, "{}-".format(testLocGenPath))
+        return testLocGenPath
+
+    async def apply_locale(self, lang):
+        """ Effectively apply the locale configuration to the new system. """
+        env = os.environ.copy()
+        localeGenPath = self._locale_gen_file_path()
+        if self._update_locale_gen_file(localeGenPath, lang) is False:
+            log.error("Failed to update locale.gen")
+            return
+
+        ok = await self._install_check_lang_support_packages(lang, env)
+
+        if ok is False:
+            log.error("Failed to install recommended language packs.")
+            return
+
+        ok = await self._activate_locale(lang, env)
+        if ok is False:
```

Comment:
ah, when I said explicitely comparison, don’t need to check boolean against True or False (as we compare them here)

Reply:
Alright.